charts/opentelemetry-demo/values.yaml

```diff
@@ -1,4 +1,7 @@
 observability:
+  # override the otel exporter endpoint to use
+  otelExporterOTLPGRPCEndpoint: ""
+  otelExporterOTLPHTTPEndpoint: ""
```

Comment:
Let's don't introduce these options. You can just override `default.env` instead

Reply:
But then I have to re-define all the existing default values in `default.env` right?

Reply:
Plus, I still have to override individual ones that require HTTP vs GRPC which is quite cumbersome.

Reply:
Correct. If it's configurable (even with more lines of code), we don't want to introduce another yaml option. Otherwise the the yaml interface blows up and becomes hard to read and maintain

Reply:
> Plus, I still have to override individual ones that require HTTP vs GRPC which is quite cumbersome, especially since the values are arrays, that means for shippingService.env and emailService.env, as well as default.env, I have to copy the whole set of existing values to my values.yaml

I'm ok with moving the env vars to default, so you will only need to override `default.env`, right?

Reply:
I think one option is to switch the env vars from a list to a dict so that they can merge properly. We don't do it anywhere in this repo, but I'd be interested to see examples of other charts using that technique.

Reply:
The problem is the fact the the merge behavior is replace for arrays, so because env-vars are in an array, it means I have to copy existing defaults, otherwise the service literally won't even run. So you can't even override knowing the defaults. That means if you change the default values, users overriding the component-level envs, won't get those updates.

Reply:
>  I'm ok with moving the env vars to default, so you will only need to override default.env, right?

No because the same env-var can't specify both HTTP and GRPC OTLP endpoints, and different applications use different values here. The email and shipping service use HTTP, while other apps use GRPC. Setting them once at the default-env won't let you do this.

Reply:
> I think one option is to switch the env vars from a list to a dict so that they can merge properly. We don't do it anywhere in this repo, but I'd be interested to see examples of other charts using that technique.

Yes, we can you an approach similar to https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/values.yaml#L186 where keys are used for overriding only

Reply:
That could work, though it would be a breaking change, and would still require more configuration on the user's side, and I feel like this would be simpler for what IMO, is likely quite a common use-case.

Reply:
I believe it's ok to make such breaking change given that it's still days of the demo chart. Better than introducing redundant configuration options

Reply:
I think there is a lot of value in being able to easily configure the collector endpoints in one place for all services. I like @TylerHelmuth's suggestion to use a map for env vars to make it easier to override specific variables, FWIW.

Reply:
Refactoring the chart to use a map for env-vars is a bigger task than I can take on right now, so I'll have to defer that.